warehouse/configuration_testing/validate.go

```diff
@@ -325,19 +343,33 @@ func (ct *CTHandleT) createTable() (err error) {
 	)
 	// Creating create table query and running over the warehouse
 	if ct.GetDestinationType() == warehouseutils.BQ {
-		bqHandle := bigquery.HandleT{}
-		err = bqHandle.CreateTestTable(&ct.client, ct.warehouse, ct.stagingTableName, TestTableSchemaMap, context.TODO())
+		bqHandle := ct.GetBigQueryHandle()
+		err = bqHandle.CreateTable(ct.stagingTableName, TestTableSchemaMap)
 	} else {
 		_, err = ct.client.Query(ct.CreateTableQuery(), client.Write)
 	}
 	return
 }
 
+func (ct *CTHandleT) alterTable() (err error) {
+	// Creating alter table query and running over the warehouse
+	if ct.GetDestinationType() == warehouseutils.BQ {
+		bqHandle := ct.GetBigQueryHandle()
+		for columnName, columnType := range AlterColumnMap {
+			err = bqHandle.AddColumn(ct.stagingTableName, columnName, columnType)
+		}
+	} else {
+		_, err = ct.client.Query(ct.AlterTableQuery(), client.Write)
+	}
```

Comment:
I am a bit skeptical here with this code design. This spreads the internals of BQ and leaks it into the code. Can we wrap it in a interface and let the handlers be the destinations. So we have one place to change the things in case something changes. Wdyt ?

Reply:
Done.